File: leetcode/algorithms/tree/binary-tree-inorder-traversal.java
/*
    leetcode
    category: "Algorithms"
    problem title: "Binary Tree Inorder Traversal"
*/

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    
    private void addToInorderTraversal(TreeNode node,List<Integer> inorderTravList){
        if(node==null){
            return;
        }
        addToInorderTraversal(node.left,inorderTravList);
        inorderTravList.add(node.val);
        addToInorderTraversal(node.right,inorderTravList);
    }
    
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> inorderTravList = new ArrayList<Integer>();
        
        addToInorderTraversal(root,inorderTravList);
        
        return inorderTravList;
        
    }
}

File: leetcode/algorithms/tree/binary-tree-inorder-traversal.ts
/*
    leetcode
    category: "Algorithms"
    problem title: "Binary Tree Inorder Traversal"
*/

/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function inorderTrav(root: TreeNode | null, trav: number[]): void {
    if(root===null){
        return;
    }
    inorderTrav(root.left,trav);
    trav.push(root.val);
    inorderTrav(root.right,trav);
}

function inorderTraversal(root: TreeNode | null): number[] {
    const trav: number[] = [];
    inorderTrav(root,trav);
    return trav;
};